src/react/honeypot-inputs.tsx

```diff
@@ -1,51 +1,58 @@
 import * as React from "react";
-import { useRouteLoaderData } from "@remix-run/react";
-import type { HoneypotInputProps } from "../server/honeypot.jsx";
+import { type HoneypotInputProps } from "../server/honeypot.js";
 
-export function HoneypotInputs() {
-  let rootLoaderData = useRouteLoaderData(
-    "root"
-  ) as Partial<HoneypotInputProps>;
+type HoneypotContextType = Partial<HoneypotInputProps>;
 
-  if (!rootLoaderData) throw new Error("Missing loader data from root");
+const HoneypotContext = React.createContext<HoneypotContextType>({});
 
-  if (!rootLoaderData.nameFieldName) {
-    throw new Error("Missing Honeypot's nameFieldName on root loader data");
-  }
+export function HoneypotInputs(): JSX.Element {
+  const context = React.useContext(HoneypotContext);
 
-  if (!rootLoaderData.validFromFieldName) {
-    throw new Error(
-      "Missing Honeypot's validFromFieldName on root loader data"
-    );
-  }
-
-  if (!rootLoaderData.encryptedValidFrom) {
-    throw new Error(
-      "Missing Honeypot's encryptedValidFrom on root loader data"
-    );
-  }
+  const {
+    nameFieldName = "name__confirm",
+    validFromFieldName = "from__confirm",
+    encryptedValidFrom,
+  } = context;
```

Comment:
Adding fallbacks for things is sensible because people still get the benefits of the honeypot even if they don't set up the context provider and things. So this is a good solution to get a quick and easy honeypot, and then opt into the more complex solution as needed.